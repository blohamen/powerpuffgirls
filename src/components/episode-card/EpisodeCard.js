import React from 'react';
import PropTypes from 'prop-types';
import styles from './EpisodeCard.module.css';
import {Link} from "react-router-dom";

export default function EpisodeCard({id, name, image}) {
    const renderName = () => (
        <span>{name}</span>
    );
    const renderImage = () => (
        <img className={styles.episodeCardImage} src={image.medium}/>
    );
    return (
        <Link to={`/episode/${id}`}>
            <div className={styles.episodeCard}>
                {image
                    ? renderImage()
                    : renderName()
                }
            </div>
        </Link>
    )
}

EpisodeCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.object,
};
