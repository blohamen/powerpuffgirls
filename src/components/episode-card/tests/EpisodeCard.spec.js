import React from 'react';

import renderer from 'react-test-renderer';
import EpisodeCard from "../";

jest.mock('react-router-dom', () => ({
    Link: 'Link',
}));

describe('Episode card component',() => {
    const episodeCardData = {
        id: 1,
        name: "Test",
        image: {
            medium: 'someUrl',
        },
    };
    it('should renders correctly with image', () => {
        const cmp = renderer.create(
            <EpisodeCard {...episodeCardData} />
        );
        expect(cmp.toJSON()).toMatchSnapshot();
    });
    it('should renders correctly only with name', () => {
        const cmp = renderer.create(
            <EpisodeCard id={episodeCardData.id} name={episodeCardData.name} />
        );
        expect(cmp.toJSON()).toMatchSnapshot();
    })
})
