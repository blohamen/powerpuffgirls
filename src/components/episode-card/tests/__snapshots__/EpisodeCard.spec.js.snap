// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode card component should renders correctly only with name 1`] = `
<Link
  to="/episode/1"
>
  <div
    className="episodeCard"
  >
    <span>
      Test
    </span>
  </div>
</Link>
`;

exports[`Episode card component should renders correctly with image 1`] = `
<Link
  to="/episode/1"
>
  <div
    className="episodeCard"
  >
    <img
      className="episodeCardImage"
      src="someUrl"
    />
  </div>
</Link>
`;
